AiTableTopGameMaster.ConsoleApp/Clients/ConsoleChatClient.cs

```diff
@@ -48,16 +50,53 @@ public async Task<ChatHistory> ChatAsync(ChatHistory history, CancellationToken
         
         try
         {
-            // Use the agent framework properly for chatting
-            await foreach (AgentResponseItem<ChatMessageContent> response in agent.InvokeAsync(history, cancellationToken: cancellationToken))
+            const int maxRevisionAttempts = 3;
+            int revisionAttempt = 0;
+            
+            while (revisionAttempt < maxRevisionAttempts)
             {
-                ChatMessageContent message = response.Message;
-                history.Add(message);
+                // Generate response from the game master agent
+                string gameMasterResponse = await GenerateGameMasterResponseAsync(history, cancellationToken);
                 
-                log.LogInformation("{Agent}: {Content}", agent.Name, message.Content);
+                // Review the response
+                OutputReviewResult reviewResult = await outputReviewer.ReviewOutputAsync(gameMasterResponse, cancellationToken);
                 
-                console.Markup($"{DisplayHelpers.AI}{agent.Name}:[/] ");
-                console.MarkupLineInterpolated($"{message.Content}");
+                if (reviewResult.IsAcceptable)
+                {
+                    // Response is acceptable, add to history and display to player
+                    ChatMessageContent acceptedMessage = new(AuthorRole.Assistant, gameMasterResponse);
+                    history.Add(acceptedMessage);
+                    
+                    log.LogInformation("{Agent}: {Content}", agent.Name, gameMasterResponse);
+                    console.Markup($"{DisplayHelpers.AI}{agent.Name}:[/] ");
+                    console.MarkupLineInterpolated($"{gameMasterResponse}");
+                    console.WriteLine();
+                    return history;
+                }
+                
+                // Response needs revision
+                revisionAttempt++;
+                log.LogDebug("Game master response needs revision (attempt {Attempt}/{Max}): {Issues}", 
```

Comment:
I'd like to see intermediate responses with failure messages as red or orange3 messages in the console as well for debugging / demo purposes.